File: src/api/java/org/models/organization/relation/Contains.java
package org.models.organization.relation;

import org.models.organization.Organization;
import org.models.organization.entity.Characteristic;
import org.models.organization.entity.Role;

/**
 * The {@linkplain Contains} interface defines the contains relation, where a {@linkplain Role} contains a {@linkplain Characteristic}, of an
 * {@linkplain Organization}.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
public interface Contains {
	/**
	 * Returns the {@linkplain Role} of this {@linkplain Contains}.
	 *
	 * @return the {@linkplain Role} of this {@linkplain Contains}.
	 */
	Role getRole();

	/**
	 * Returns the {@linkplain Characteristic} of this {@linkplain Contains}.
	 *
	 * @return the {@linkplain Characteristic} of this {@linkplain Contains}.
	 */
	Characteristic getCharacteristic();

	/**
	 * Returns the <code>double</code> value of this {@linkplain Contains}.
	 *
	 * @return the <code>double</code> value of this {@linkplain Contains}.
	 */
	double getValue();

	/**
	 * Sets the value of this {@linkplain Contains} to the given <code>double</code> value.
	 *
	 * @param value
	 *            the <code>double</code> value to set.
	 */
	void setValue(double value);
}

File: chazm-api/src/main/kotlin/runtimemodels/chazm/api/relation/Contains.kt
package runtimemodels.chazm.api.relation

import runtimemodels.chazm.api.entity.Characteristic
import runtimemodels.chazm.api.entity.Role
import runtimemodels.chazm.api.organization.Organization

/**
 * The [Contains] interface defines a contains relation, where a [Role] contains a [Characteristic], of an
 * [Organization].
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
interface Contains {
    /**
     * The [Role] of this [Contains].
     */
    val role: Role

    /**
     * The [Characteristic] of this [Contains].
     */
    val characteristic: Characteristic

    /**
     * The `double` value of this [Contains].
     */
    var value: Double
}
